types/spec/signal.d.ts

```diff
@@ -12,9 +12,10 @@ export interface PushSignal extends BaseSignal {
   push?: 'outer';
 }
 export interface NewSignal extends BaseSignal {
-  value?: any;
+  value?: SignalValue;
   react?: boolean;
   update?: Expr;
   bind?: Binding;
 }
 export type Signal = NewSignal | PushSignal;
+export type SignalValue = boolean | Date | number | [number, number] | string | undefined;
```

Comment:
Hm, why is a signal value constrained to these types? In my code I assign arbitrary objects e.g. `{ value: { foo: 42, bar: 'hello' } }` and it works perfectly fine, but with these new typings I suddenly have a type error.

Reply:
Hi @pelotom - I should have said that these are just a starting point. I actually did not know the entire range of values. Should we revert to `any`? Or can you add more types to this list?

Reply:
AFAIK there's no useful limitation we can put on the value of a signal, so it should just be `any`. Would you agree @domoritz?

Reply:
see https://github.com/vega/vega-typings/pull/52

Reply:
Hm. I don't see how we can enumerate all the possible values that are allowed. We could add `{}` to this list, but at that point we might as well just define it as `unknown`. Is `null` not allowed?

Reply:
It feels like `any` is correct. From a typing perspective, perhaps we can go with a generic? Example: `SignalValue<T>`

Reply:
I think a generic would be more trouble than it's worth. It would have to bubble up to the level of the `View`, so you'd have essentially `View<S1, S2, S3>` where `S1`, `S2`, `S3` are all the signals being used in the view. The marginal type safety gained at the cost of this verbosity seems dubious.

Reply:
Let's revert to `any`. Why `SignalValue<T>`>?

Reply:
And at the end of the day you can't really enforce type safety anyway, since the majority of the interaction with signals happens within opaque string expressions in the spec.

Reply:
Correct - ouch. Sending a new PR with `any` 👍

Reply:
sent #56

Reply:
Merged and released.